saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala

```diff
@@ -288,6 +288,29 @@ object MultiModalSpRLDataModel extends DataModel {
       pos(first) + "::" + pos(second)
   }
 
+  val imageConfirmsRelation = property(pairs, cache = true) {
+    var result = false
+    r: Relation =>
+      val (x, rel) = getArguments(r)
+      val img = ((pairs(r)<~sentenceToRelations<~documentToSentence)~>documentToImage).head
+      val seg = images(img)~> imageToSegment
+
+      val imageRelations = segmentRelations.getAllInstances.filter(sr=> sr.getImageId().equals(img.getId()))
```

Comment:
you are not allowed to use `getAllInstances` because we don't know if we are in test or train phase and `getAllInstances` returns both test and train instances.

use `~>` or `<~` to traverse the graph.
`imageRelations =  images(img)~> imageToSegment<~ segmentRelationsToSegments`